client/state/jetpack/credentials/reducer.js

```diff
@@ -63,6 +66,30 @@ export const updateRequestStatus = keyedReducer( 'siteId', ( state, { type } ) =
 	return state;
 } );
 
+export const testRequestStatus = keyedReducer( 'siteId', ( state, { type, role } ) => {
+	switch ( type ) {
+		case JETPACK_CREDENTIALS_TEST:
+			return {
+				...state,
+				[ role ]: 'pending',
```

Comment:
Another idea for improvement here is to define constants with possible states like `const JETPACK_CREDENTIALS_PENDING = 'pending'` and reusing that everywhere. While that seems like a very slight improvement, it can help with reducing the potential for errors and could make the code more intuitive. In cases where those are used a lot, it can also help with code organization.